gcc is broken after #7363

#7363 broke my gcc (5.4 and 6.2):

> /users/ag/lib/dealii-master/source/base/subscriptor.cc:203:75: error: no matching function for call to 'find(std::vector<bool*>::iterator, std::vector<bool*>::iterator, bool* const&)'
>      std::find(validity_pointers.begin(), validity_pointers.end(), validity);
>                                                                            ^
> In file included from /opt/gcc/6.2.0/snos/include/g++/bits/locale_facets.h:48:0,
>                  from /opt/gcc/6.2.0/snos/include/g++/bits/basic_ios.h:37,
>                  from /opt/gcc/6.2.0/snos/include/g++/ios:44,
>                  from /opt/gcc/6.2.0/snos/include/g++/istream:38,
>                  from /opt/gcc/6.2.0/snos/include/g++/sstream:38,
>                  from /opt/gcc/6.2.0/snos/include/g++/complex:45,
>                  from /users/ag/lib/dealii-master/include/deal.II/base/numbers.h:29,
>                  from /users/ag/lib/dealii-master/build/include/deal.II/base/config.h:425,
>                  from /users/ag/lib/dealii-master/include/deal.II/base/logstream.h:19,
>                  from /users/ag/lib/dealii-master/source/base/subscriptor.cc:16:
> /opt/gcc/6.2.0/snos/include/g++/bits/streambuf_iterator.h:369:5: note: candidate: template<class _CharT2> typename __gnu_cxx::__enable_if<std::__is_char<_CharT2>::__value, std::istreambuf_iterator<_CharT> >::__type std::find(std::istreambuf_iterator<_CharT>, std::istreambuf_iterator<_CharT>, const _CharT2&)
>      find(istreambuf_iterator<_CharT> __first,
>      ^~~~
> /opt/gcc/6.2.0/snos/include/g++/bits/streambuf_iterator.h:369:5: note:   template argument deduction/substitution failed:
> /users/ag/lib/dealii-master/source/base/subscriptor.cc:203:75: note:   '__gnu_cxx::__normal_iterator<bool**, std::vector<bool*> >' is not derived from 'std::istreambuf_iterator<_CharT>'
>      std::find(validity_pointers.begin(), validity_pointers.end(), validity);
>                                                                            ^

--- a/source/base/subscriptor.cc
+++ b/source/base/subscriptor.cc
@@ -17,6 +17,7 @@
 #include <deal.II/base/subscriptor.h>
 #include <deal.II/base/thread_management.h>
 
+#include <algorithm>
 #include <iostream>
 #include <string>
 #include <typeinfo>

